linux/LinuxProcessList.c

```diff
@@ -716,22 +723,78 @@ static bool LinuxProcessList_readCmdlineFile(Process* process, const char* dirna
       return false;
    }
    for (int i = 0; i < amtRead; i++) {
-      if (command[i] == '\0' || command[i] == '\n') {
-         if (tokenEnd == 0) {
+      /* newline used as delimiter - when forming the mergedCommand, newline is
+       * converted to space by LinuxProcess_makeCommandStr */
+      if (command[i] == '\0')
+         command[i] = '\n';
+
+      if (command[i] == '\n') {
+         if (tokenEnd == 0)
             tokenEnd = i;
-         }
-         command[i] = ' ';
       } else {
+         /* htop considers the next character after the last / that is before
+          * basenameOffset, as the start of the basename in cmdline - see
+          * Process_writeCommand */
+         if (!tokenEnd && command[i] == '/')
+            tokenStart = i + 1;
          lastChar = i;
       }
```

Comment:
Please use braces, even where they are optional. Readability of the source will be better.

Reply:
ok.